### DISCUSSION TITLE
Adding and deleting values into field array is resetting the data.

### DISCUSSION THREAD

**Discussion by vejandla** [react-hook-form/react-hook-form#5516] [Q&A]
Hi, 

I'm using RHF 7.0 and used from provider in my app. 

I have a situation where a nested field array has a dropdown element in it and everytime user selects a dropdown, then that option should not be visible in the next row of that field array. Since RHF does not re-render, I used useWatch to make the nested array component to rerender. 

Upon every addition or removal of the element into the field array, it resets the user entered values in other rows.

Appreciate any help. Here is CSB link.

https://codesandbox.io/s/gracious-gould-v5ff9?file=/src/VehicleContainer.js

**While deleting, it is always deleting the first index even though the index sending to remove function is different.**

**Comment by bluebill1049** [comment]
can you isolate your example with so many components? i will take a look at it tmr.

**Reply by vejandla** [reply]
@bluebill1049 Thanks for the update. I've skimmed down some of the components that can be avoided.

**Reply by vejandla** [reply]
Hi @bluebill1049 did you get a chqnce to look into it? We got struck and couldn't move forward. Appreciate your help with this.

**Reply by bluebill1049** [reply]
hey i haven't got a chance to look into your issue, because the CSB has quite a lot of dep still, but I quickly did one codesandbox with a nested field array, hopefully, by following this you will find out the root cause.

https://codesandbox.io/s/vehicle-n8c9k?file=/src/index.js

**Reply by vejandla** [reply]
thanks, @bluebill1049.  I'll compare your example with mine. But on quick compare here is what I found. 

1) I'm using `form provider` and your example is not. 
2) You are not using useWatch. Until I use the `useWatch` things were normal. I was trying to force a render as If the user selects a value in the nested dropdown/select then that value should not appear for the next element in the nested array.
3) Lastly, Do you mean the dependencies are Material UI or too many files? If it is too many files, I will dump everything into one file if it makes it easier and also will remove `yup` related stuff too.

**Reply by bluebill1049** [reply]
I will include useWatch, provider doesn't make any difference. Show me the file which you are using useWatch.

**Reply by vejandla** [reply]
It's in vehicle.js 
https://codesandbox.io/s/gracious-gould-v5ff9?file=/src/Vehicle.js

**Reply by bluebill1049** [reply]
<img width="678" alt="Screen Shot 2021-06-14 at 10 12 08 pm" src="https://user-images.githubusercontent.com/10513364/121890343-91668600-cd5d-11eb-9c28-9f85d7eca084.png">

- This component is getting unmount and remount each render

**Reply by bluebill1049** [reply]
I think that could be the root cause, let me know if this fix the problem or not.

**Reply by bluebill1049** [reply]
sorry for getting you late on the reply... just been super busy over the week... I do appreciate your patient by politely ping me back.

**Reply by vejandla** [reply]
I appreciate your help. Thanks for all the help you are doing to the community. It is invaluable. 🥇
